File: src/equiv/TrailingZeroes4.ml
module JB = Javalib_pack.JBasics
module QID = QualifiedIdentity
module E = Expr

let _ =
  let x = E.Var (Var.Mk (QID.QID ["x"], Type.Int)) in

  Equiv.run "TrailingZeroes4"
            [(JB.TBasic `Int)]
            (JB.TBasic `Int)
            "trailingZeroes2"
            "trailingZeroes4"
            [x]
            [E.mk_ige x (E.Int 5)]

File: benchmark/equiv/TrailingZeroes4.java
public class TrailingZeroes4 {
  static public int trailingZeroes2(int n) {
    int sum = 0;
    while (n >= 5) {
      sum += n / 5;
      n = n / 5;
    }
    return sum;
  }

  static public int trailingZeroes4(int n) {
    int x = 0;
    int y = n / 5;
    while (y!=0) {
      x = x + y;
      y = y / 5;
    }
    return x;
  }
}
